kong/plugins/aws-lambda/handler.lua

```diff
@@ -38,6 +38,51 @@ local server_header = meta._SERVER_TOKENS
 local AWS_PORT = 443
 
 
+--[[
+  Response format should be
+  {
+      "statusCode": httpStatusCode,
+      "headers": { "headerName": "headerValue", ... },
+      "body": "..."
+  }
+--]]
+local function validate_custom_response(response)
+  if type(response.statusCode) ~= "number" then
+    return nil, "statusCode must be a number"
+  end
+
+  local headers = response.headers
+  local body = response.body
+  if headers ~= nil and type(headers) ~= "table" then
+    return nil, "headers must be a table"
+  end
+
+  if body ~= nil and type(response.body) ~= "string" then
+    return nil, "body must be a string"
+  end
+
+  return true
+end
+
+
+local function extract_proxy_response(content)
+  local serialized_content = cjson.decode(content)
+  local ok, err = validate_custom_response(serialized_content)
+  if not ok then
+    return nil, err
+  end
+
+  local headers = serialized_content.headers or {}
+  local body = serialized_content.body or ""
+  headers["Content-Length"] = #body
+
+  return {
+    status_code = tonumber(serialized_content.statusCode),
```

Comment:
Please cache the `tonumber` global lookup at the beginning of this file, just like other globals that are accessed :)